homeassistant/components/bmw_connected_drive/__init__.py

```diff
@@ -28,6 +34,7 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+DATA_HASS_CONFIG = "bmw_connected_drive_hass_config"
```

Comment:
Please use the `DOMAIN` as the main key in `hass.data` for this integration. Then we can nest more keys and dicts under that key.

Reply:
I only added this to store the HA config from `async_setup` as the tibber integration does. 

In `async_setup_entry`, I am using `hass.data.setdefault(DOMAIN, {})`, as multiple configs can be created and to store the single entities as dictionary items below `hass.data[DOMAIN]`. If I understand correctly, also putting the `config` from `async_setup` to `hass.data[DOMAIN]` doesn't make sense? I only want the *parsed* data, i.e. the `ConfigEntries` in `hass.data[DOMAIN]`, right?
I could imagine storing the HASS `config` from `async_setup` in `hass.data[DOMAIN]['hass_config']`, but then I would always need to double check and not parse this as a valid `ConfigEntry`.

Additionally, it would differ from the *tibber* integration you mentioned as a reference a couple of times.

Reply:
Please move all the integration data to be stored in `hass.data` under the top key `DOMAIN`. Put a dictionary there and then put as many sub keys and dicts as needed in there.

Reply:
Done, thanks for the clarification!